### PR TITLE
(DOCSP-20092): [Swift SDK] Delete user documentation

### PR DESCRIPTION
**Pull request by dacharyc**
### Jira

- https://jira.mongodb.org/browse/DOCSP-20092

### Staged Changes (Requires MongoDB Corp SSO)

- [Create and Delete Users (Swift SDK)](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-20092/sdk/swift/examples/users/create-and-delete-users/): New page w/new code examples for deleting the user
- [Work with Users/Create and Delete Users](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-20092/sdk/swift/examples/work-with-users/#create-and-delete-users): New section on existing page
- [Create a User (Cloud)](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-20092/users/create/): Added a section about first authentication automatically creating user objects, and a link to SDK pages where devs can create email/password users
- [Delete a User (Cloud)](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-20092/users/delete-or-revoke/#delete-a-user-in-the-sdk): New section with a link to the Swift SDK where you can delete a user via SDK. As more SDKs add the user.delete API, we can add tabs

---

### Before merging:
- [x] Update `..versionadded` to reflect the release version
- [x] Update Podfile to point to release version
- [x] Add link from ``delete`` to generated SDK reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: examples/ios/Examples/Authenticate.swift [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import XCTest
-import RealmSwift
-import GoogleSignIn
-import FBSDKLoginKit
+ import RealmSwift
+ import GoogleSignIn
+ import FBSDKLoginKit
 
-class Authenticate: XCTestCase {
+ class Authenticate: XCTestCase {
```
Comment (cbush) [inline examples/ios/Examples/Authenticate.swift:6]: Why these changes?

File: examples/ios/Examples/DeleteUsers.swift [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                // :hide-start:
+                loginExpectation.fulfill()
+                // :hide-end:
+            }
+        }
+        // :hide-start:
+        wait(for: [loginExpectation], timeout: 10)
+        // :hide-end:
+
+        // Now we have a user, and the total users in the app = 1
```
Comment (cbush) [inline examples/ios/Examples/DeleteUsers.swift:27]: This doesn't really work as a code example since the blocking call to wait() is hidden. You could get around this with a comment like "// Later, after the user is logged in..." or put the continuation in the continuation of the login call.

File: examples/ios/Examples/DeleteUsers.swift [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        XCTAssertNotNil(syncUser)
+        XCTAssertEqual(app.allUsers.count, 1)
+
+        // :hide-start:
+        let deleteExpectation = XCTestExpectation(description: "User deleted")
+        // :hide-end:
+        // Call the `delete` method to delete the user
+        syncUser?.delete { (error) in
```
Comment (cbush) [inline examples/ios/Examples/DeleteUsers.swift:35]: We've already asserted the syncUser so we should force unwrap it (if we need to unwrap it at all?)

File: source/includes/note-delete-user.rst [hunk 4]
```diff
@@ -0,0 +1,6 @@
+.. note::
+
+   Realm does not automatically delete any data in your :term:`linked {+atlas+}
+   cluster <linked cluster>` that you have associated with a deleted user, such
+   as by including their ID in an ``owner_id`` field. To remove all traces of a
```
Comment (cbush) [inline source/includes/note-delete-user.rst:5]: "such as by" feels garbled. Could you clarify?

File: source/sdk/swift/examples/users/create-and-delete-users.txt [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+the user from your Realm application.
+
+.. important::
+
+   Deleting a user only deletes the user object, which may contain associated 
+   :ref:`metadata <ios-user-metadata>`. This does not delete :ref:`custom 
+   user data <ios-custom-user-data>` or user-entered data from your 
+   application. Apple :apple:`requires that you disclose data retention 
+   and deletion policies <app-store/review/guidelines/#5.1.1>` to your 
+   application customers, and give them a way to request user data deletion.
```
Comment (cbush) [inline source/sdk/swift/examples/users/create-and-delete-users.txt:47]: "Requires that you X and Y" (no subject change ("you") = no comma)
```suggestion
   application customers and give them a way to request user data deletion.
```

File: source/sdk/swift/examples/users/create-and-delete-users.txt [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+   If you collect additional user data, you must handle deleting that data.
```
Comment (cbush) [inline source/sdk/swift/examples/users/create-and-delete-users.txt:48]: Maybe? I think there is a certain nuance in "handle" here:
```suggestion
   If you collect additional user data, you must delete that data.
```

**Review by cbush — APPROVED** [review]

File: examples/ios/Examples/Authenticate.swift (hunk 1 above)
Comment (dacharyc) [reply examples/ios/Examples/Authenticate.swift:6]: I did not change the spacing (intentionally) so I'm not sure why that has changed. But I did add the user.delete call below in ln 144 because it was causing the DeleteUsers `.count` tests to fail.

File: examples/ios/Examples/DeleteUsers.swift (hunk 3 above)
Comment (dacharyc) [reply examples/ios/Examples/DeleteUsers.swift:35]: Compiler requires unwrapping, but I'll swap it to a force unwrap.

File: source/sdk/swift/examples/users/create-and-delete-users.txt (hunk 6 above)
Comment (dacharyc) [reply source/sdk/swift/examples/users/create-and-delete-users.txt:48]: You're right, there is a nuance here, and I'm not sure `delete` really covers it - what about:

> If you collect additional user data, you must implement your own methods or processes to delete that data.

Comment (cbush) [reply source/sdk/swift/examples/users/create-and-delete-users.txt]: Yes, I like that.

File: source/sdk/swift/examples/users/create-and-delete-users.txt [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+.. literalinclude:: /examples/generated/code/start/DeleteUsers.codeblock.async-await-delete-user.swift
+   :language: swift
+
+Delete Users with Completion Handlers
+-------------------------------------
+
+If your application does not use async/await:
+
+.. literalinclude:: /examples/generated/code/start/DeleteUsers.codeblock.closure-delete-user.swift
+   :language: swift
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/sdk/swift/examples/users/create-and-delete-users.txt:67]: What's really weird is that this new line at the end of the file exists on my local. I don't understand!
